lib/sparrow/h2_worker.ex

```diff
@@ -52,16 +53,32 @@ defmodule Sparrow.H2Worker do
       end
 
     state = Sparrow.H2Worker.State.new(nil, config)
+
+    :telemetry.execute(
+      [:sparrow, :h2_worker, :init],
+      %{},
+      extract_worker_info(state)
+    )
+
     {:ok, state, {:continue, :start_conn_backoff}}
   end
 
   @spec terminate(reason, state) :: :ok
-  def terminate(reason, %State{connection_ref: nil}) do
+  def terminate(reason, state = %State{connection_ref: nil}) do
     _ =
       Logger.info(fn ->
         "action=terminate, reason=#{inspect(reason)}, connection_ref=nil"
       end)
 
+    :telemetry.execute(
+      [:sparrow, :h2_worker, :terminate],
+      %{},
+      state
+      |> extract_worker_info()
+      |> Map.put(:reason, reason)
+      |> Map.put(:requests, state.requests)
```

Comment:
`requests` is a big structure with complex structure, thus should not be used as metadata.